x-pack/plugins/ml/public/application/management/jobs_list/components/jobs_list_page/space_management/space_management.tsx

```diff
@@ -0,0 +1,193 @@
+import { getColumns } from './columns';
+import { MLSavedObjectsSpacesList } from '../../../../../components/ml_saved_objects_spaces_list';
+import { getFilters } from './filters';
+
+interface Props {
+  spacesApi: SpacesPluginStart | undefined;
+  setCurrentTab: (tabId: MlSavedObjectType) => void;
+}
+
+export const SpaceManagement: FC<Props> = ({ spacesApi, setCurrentTab }) => {
+  const { getList } = useManagementApiService();
+  const [currentTabId, setCurrentTabId] = useState<MlSavedObjectType>('anomaly-detector');
+  const [jobs, setJobs] = useState<ManagementListResponse>();
+  const [columns, setColumns] = useState<Array<EuiBasicTableColumn<ManagementItems>>>([]);
+  const [filters, setFilters] = useState<SearchFilterConfig[] | undefined>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const refresh = useCallback(() => {
+    setIsLoading(true);
+    getList(currentTabId)
+      .then((jobList) => {
+        setJobs(jobList);
+        setIsLoading(false);
+        setFilters(getFilters(currentTabId, jobList));
+      })
+      .catch(() => {
+        setJobs([]);
+        setFilters(undefined);
+        setIsLoading(false);
+      });
+  }, [getList, currentTabId]);
+
+  useEffect(() => {
+    setJobs(undefined);
+    setColumns(createColumns());
+    setCurrentTab(currentTabId);
+    refresh();
+  }, [currentTabId]);
+
+  const createColumns = useCallback(() => {
+    return [
+      ...getColumns(currentTabId),
+      ...(spacesApi !== undefined
+        ? [
+            {
+              name: i18n.translate('xpack.ml.management.spaceManagementTableColumn.spaces', {
+                defaultMessage: 'spaces',
+              }),
+              'data-test-subj': 'mlSpaceManagementTableColumnSpaces',
+              sortable: true,
+              truncateText: true,
+              render: (item: ManagementItems) => {
+                return (
+                  <MLSavedObjectsSpacesList
+                    spacesApi={spacesApi}
+                    spaceIds={item.spaces ?? []}
+                    id={item.id}
+                    mlSavedObjectType={currentTabId}
+                    refresh={refresh}
+                  />
+                );
+              },
+            },
+          ]
+        : []),
+    ] as Array<EuiBasicTableColumn<ManagementItems>>;
+  }, [currentTabId, spacesApi, refresh]);
+
+  const getTable = useCallback(() => {
+    return (
+      <>
+        <EuiSpacer size="m" />
+        {jobs === undefined ? null : (
+          <>
+            <EuiFlexGroup>
+              <EuiFlexItem />
+              <EuiFlexItem grow={false}>
+                <RefreshButton onRefreshClick={refresh} isRefreshing={isLoading} />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+            <EuiInMemoryTable
+              data-test-subj={`mlSpacesManagementTable-${currentTabId} ${
+                isLoading ? 'loading' : 'loaded'
+              }`}
+              items={jobs as any}
```

Comment:
did you try providing a generic for `EuiInMemoryTable`? would be great to avoid using `any`

Reply:
yes, I meant to change this.